DMCompiler/DM/Visitors/DMObjectBuilder.cs

```diff
@@ -94,13 +95,18 @@ class DMObjectBuilder {
                 } else {
                     DMVariable variable = new DMVariable(null, varOverride.VarName, false);
 
+                    if (variable.Value is not null && variable.Value.ValType == DMValueType.Unimplemented)
+                    {
+                        Program.Warning(new CompilerWarning(null, $"{variable.Type}.{variable.Name} is not implemented and will have unexpected behavior"));
```

Comment:
`variable.Type` is always null here. The way that variable overrides currently work, we don't know anything about the variable being overriden.